client/src/www/app/main.ts

```diff
@@ -52,12 +51,13 @@ function getRootEl() {
 function createServerRepo(
   eventQueue: EventQueue,
   storage: Storage,
-  deviceSupport: boolean,
-  createTunnel: TunnelFactory
 ) {
-  const repo = new OutlineServerRepository(createTunnel, eventQueue, storage);
-  if (!deviceSupport) {
+  try {
+    return new OutlineServerRepository(eventQueue, storage);
+  } catch {
     console.debug('Detected development environment, using fake servers.');
```

Comment:
This is a bit odd now - any error means we're in a development environment? It would be clearer catching a  specific error for that instead of catching all possible errors: `if (e instanceof SpecificError)`.

Reply:
Maybe there could be some more potential refactors, by moving development environment (i.e., browser) to a new `main.browser.ts`.

Reply:
Makes sense. I changed it so that it returns undefined if not supported, and I check accordingly.